tests/tests/test-staking/test-staking-locks.ts

```diff
@@ -543,11 +537,19 @@ describeDevMoonbeam("Staking - Locks - bottom and top delegations", (context) =>
       context.createBlock(
         [...topDelegators].map((account, i) =>
           context.polkadotApi.tx.parachainStaking
-            .delegate(alith.address, MIN_GLMR_DELEGATOR + 1n * GLMR, i + 1, 1)
+            .delegate(alith.address, MIN_GLMR_DELEGATOR + 1n * GLMR, 9999, 1)
             .signAsync(account)
         )
       )
     );
+
+    // XXX: wtf?
+    await context.createBlock();
+    await context.createBlock();
+    await context.createBlock();
+    await context.createBlock();
+    await context.createBlock();
+
```

Comment:
This is really surprising. I think there are a couple issues going on (see code above this for more context).

(Sorry, this probably needs way more context to make sense...)

1. These txns now have identical priorities (all 0). Previously the weight hint (the `i + 1` part) was probably causing each successive txn to have slightly more weight. I would expect that to cause the txpool to prefer larger weight hints, causing this to work backwards from the "most expensive" (higher weight hint) to the least (lowest weight hint). So it's clear why changing the weight hint to `9999` fixes this, but I'm not sure why it worked before.

2. Replacing the weight hint (to `9999`) undoes the issue with #1 -- now all txns should pass because order doesn't matter. But we end up needing to make several more blocks afterward for all delegation requests to be processed. Previously, they were consistently handled in a single block. It would seem that setting all txns to the same priority has a negative effect on the txpool's ability to drain itself.